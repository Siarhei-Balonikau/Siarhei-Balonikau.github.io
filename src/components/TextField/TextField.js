import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';


const TextField = props => {
  return (
    <input value={props.value} className="text-field" type="text" onChange={(e) => props.onChange(props.field, e.target.value)} />
  );
};

TextField.propTypes = {
  onChange: PropTypes.func,
  field: PropTypes.string,
  value: PropTypes.string
};

export default TextField;