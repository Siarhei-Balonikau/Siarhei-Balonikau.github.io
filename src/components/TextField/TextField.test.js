import React, { Component } from 'react';
import { shallow } from 'enzyme';
import TextField from './TextField'; 
import toJson from 'enzyme-to-json'

describe('subscribe component', () => {
  let wrapper, props;

  beforeEach(() => {
    props = {

    };
  
    wrapper = shallow(<TextField {...props} />);
  });
  
  it('should render component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});