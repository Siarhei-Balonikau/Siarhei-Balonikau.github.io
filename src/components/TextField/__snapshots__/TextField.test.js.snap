// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subscribe component should render component 1`] = `
<input
  className="text-field"
  onChange={[Function]}
  type="text"
/>
`;
